zomes/deepkey_csr/src/key_registration.rs

```diff
@@ -15,10 +15,12 @@ use hdk_extensions::{
 
 
 #[hdk_extern]
-pub fn next_derivation_details(installed_app_id: String) -> ExternResult<DerivationDetailsInput> {
+pub fn next_derivation_details(input: Option<ByteArray<32>>) -> ExternResult<DerivationDetailsInput> {
```

Comment:
I was thinking the param would be an optional app_index but I think this works just as well.

Reply:
I figured this way was better so you don't have to track any mapping outside of deepkey.  Is there any value in supporting both ways?

Reply:
It's hard to say, I'm thinking through the possibility of a case where we want to bind additional apps to a key after the initial registration, which I think Holo may need. In that case we may need to refer to app bindings separately (or maybe not, if we just update the original). But in any case that doesn't affect derivation details, since even if there were multiple app bindings, they would all share the same derivation path and the same key. (But there might be a need for different apps to update their keys at different times even though they share the same derivation path.)